apps/nerves_hub_web_core/lib/nerves_hub_web_core/repo.ex

```diff
@@ -27,4 +29,20 @@ defmodule NervesHubWebCore.Repo do
   end
 
   def reload_assoc({:error, changeset}, _), do: {:error, changeset}
+
+  def soft_delete(struct_or_changeset) do
+    struct_or_changeset
+    |> soft_delete_changeset()
+    |> update()
+  end
+
+  def soft_delete_changeset(struct_or_changeset) do
+    deleted_at = DateTime.truncate(DateTime.utc_now(), :second)
+
+    Ecto.Changeset.change(struct_or_changeset, deleted_at: deleted_at)
+  end
+
+  def with_undeleted(query) do
```

Comment:
```suggestion
  def exclude_deleted(query) do
```
This is a fairly big nit, but "undeleted" makes me think think records were once deleted, but now are not, and we're doing complex query to deduce that.

I don't know if my suggestion is the best function name alternative, but I think something a little simpler for what it is doing would be nice.

Reply:
yep, i like that better. the thing i worried about with this was where it was placed. It's not really a repo thing it's a query thing. I thought about putting the same function in each module that would be soft deleted but then settled on this. 🤷